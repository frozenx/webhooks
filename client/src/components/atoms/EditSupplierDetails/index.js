import React, { Component } from 'react';

import Button from '@beans/button';
import Input from '@beans/input';
import Label from '@beans/label';
import {
    SupplierDetailsContainer,
    SupplierHeader,
    SupplierInfo
} from '../SupplierDetails/style'

export class EditSupplierDetails extends Component {
    
    constructor(props){
        super(props);
        this.state = {};
    }

    normalCase = (string) => {
        let result = string.replace( /([A-Z])/g, " $1" );
        let finalResult = result.charAt(0).toUpperCase() + result.slice(1);
        return finalResult;
    }
    
    render(){
        return(
            <SupplierDetailsContainer >
                <form onChange  ={(e) => this.props.handleSupplierChange(e)}>
                <SupplierHeader>
                    <h2>Supplier Details</h2>
                    <Button
                        variant = "primary"
                        className = "submit-btn"
                        onClick={this.props.handleSubmitSupplierDetails}
                        type = "submit"> Submit
                    </Button>
                    <Button
                        variant = "primary"
                        className = "cansel-btn"
                        onClick={this.props.handleCanselSupplierDetails}
                        type = "submit"> Cancel
                    </Button>
            </SupplierHeader>
            <div>
            {
                this.props.supplierData.map((attrGrp, i) => {
                    return (
                        <SupplierInfo key={i}><Label>{this.normalCase(attrGrp)}</Label>
                            <div> 
                                <Input
                                    type = "text" 
                                    id={i}
                                    defaultValue={this.props.editSupplierData[attrGrp] || ""}
                                    name = {attrGrp}
                                />
                            </div>
                        </SupplierInfo>
                    )                                         
                })
            }
            </div>
            </form>
            </SupplierDetailsContainer>
        )
    }
}

export default EditSupplierDetails;