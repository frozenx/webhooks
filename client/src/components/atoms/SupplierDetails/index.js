import React from 'react';
import Label from '@beans/label';
import {
    SupplierDetailsContainer,
    SupplierHeader,
    SupplierInfo

} from './style'

const SupplierDetails = props => {

    const {
        supplierData,
        editSupplierData,
        handleEditSupplierDetails
    } = props;

    const normalCase = (string) => {
        let result = string.replace( /([A-Z])/g, " $1" );
        let finalResult = result.charAt(0).toUpperCase() + result.slice(1);
        return finalResult;
    }
    
    return (
        <SupplierDetailsContainer>
            <SupplierHeader>
                <h2>Supplier Details</h2>
                <div className="edit-supplier" onClick={handleEditSupplierDetails}>Edit</div>
            </SupplierHeader>
            <div>
            {
                supplierData.map((attrGrp, i) => {
                   
                        return (
                            <SupplierInfo key={i}><Label> {normalCase(attrGrp)}</Label><span>{editSupplierData[attrGrp]}</span></SupplierInfo>
                        )                                         
                })
            }
            </div>
        </SupplierDetailsContainer>
    ) 
}


export default SupplierDetails;