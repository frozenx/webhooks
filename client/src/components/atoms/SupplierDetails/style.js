import styled from 'styled-components'

export const SupplierDetailsContainer = styled.div`
    background-color: #f6f6f6;
    border:1px solid #ccc;
    margin-top:30px;
    padding: 30px 20px 20px 20px;
`;

export const SupplierHeader = styled.div`
    margin-bottom: 20px;
    > h2{
        display:inline;
        font-size: 20px;
        margin-right: 10px;
    }
    button{
        margin-right: 10px;
    }
    > div{
        color: #007eb3;
        display: inline;
        margin-left: 10px;
        font-size: 16px;
        font-weight: 600;       
    }
`;

export const SupplierInfo = styled.div`
    display: inline-block;
    margin-bottom: 15px;
    width: 50%;
    font-size: 16px;
    > label{
        width: 48%;
        color: #282c34;
        font-weight: bold;
        vertical-align: top;
    }
    > span{
        color: #999;
        text-overflow: ellipsis;
        width: 200px;
        overflow: hidden;
        white-space: nowrap;
        cursor: pointer;
    }
    > div{
        cursor: pointer;
        width: 40%;
        display: inline-block;
    }
`;
