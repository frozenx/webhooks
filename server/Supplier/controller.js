const SupplierService = require('./service');
const AddressService = require('../addressBook/service');
const ContactService = require('../contactBook/service');
const SuppliersModel = require('./models/Suppliers');
const SupplierAttributes = require('./models/SupplierAttributes');
const SiteAttributes = require('./models/SiteAttributes');
const PaymentAttributes = require('./models/PaymentAttributes');
const AddressModel = require('../addressBook/models/addressAttributes');
const logger = require('../lib/logger')(module);
const MetaDataService = require('../metadata/service');
const ErrorBuilder = require('../lib/error-handlers/ErrorBuilder');
const constants = require('../constants');

class SupplierController {
  constructor(req, res) {
    this.request = req;
    this.response = res;
    this.body = req.body;
    this.params = req.query;
    this.accessToken = req.cookies['OAuth.AccessToken.EP'];
    this.supplierService = new SupplierService(this.accessToken);
    this.addressService = new AddressService(this.accessToken);
    this.contactService = new ContactService(this.accessToken);
    this.metaData = new MetaDataService(this.accessToken);
  }

  async readSupplier() {
    try {
      const isFullText = false;
      const supplierServiceResponse = await this.supplierService.readSupplier(
        this.params,
        isFullText,
      );
      const suppliersModel = new SuppliersModel(supplierServiceResponse);
      this.response.send(JSON.stringify(suppliersModel));
    } catch (error) {
      const controllerError = new ErrorBuilder('controller', error);
      logger.info(controllerError.message);
      this.response.status(controllerError.code);
      this.response.end();
    }
  }

  async readAndCreate() {
    try {
      const isFullText = true;
      const {
        companyRegNumber,
      } = this.body;

      const getCompanyAddressResponse = await this.addressService.getCompanyAddress(companyRegNumber);
      const searchSupplierAndCreateResponse = await this.supplierService.readAndCreate({
        ...this.body,
        partnerName: getCompanyAddressResponse.company_name,
      },
        isFullText,
      );
      let model;
      if (searchSupplierAndCreateResponse.uuid) { // create response
        model = new SupplierAttributes(searchSupplierAndCreateResponse);
        const addressPayload = new AddressModel().mapCompanyHouseAttributesToAddressPayload(getCompanyAddressResponse);
        await this.addressService.saveAddress({
          attributes: {
            ...addressPayload,
          },
          partnerUuid: model.uuid,
        },
          addressPayload);
      } else { // search response
        model = new SuppliersModel(searchSupplierAndCreateResponse);
      }
      this.response.send(JSON.stringify(model));
    } catch (error) {
      const controllerError = new ErrorBuilder('controller', error);
      logger.info(controllerError.message);
      if (error.isCompanyAddressError) {
        this.response.send({
          errorKeys: [constants.COMPANY_REG_NUMBER],
        });
      } else {
        this.response.status(controllerError.code);
        this.response.end();
      }
    }
  }

  async createSupplier() {
    try {
      const supplierResponse = await this.supplierService.createSupplier(
        this.body,
      );
      const supplierModel = new SupplierAttributes(supplierResponse);
      this.response.json(supplierModel);
    } catch (error) {
      const controllerError = new ErrorBuilder('controller', error);
      logger.info(controllerError.message);
      this.response.status(controllerError.code);
      this.response.end();
    }
  }

  async getSupplierDetails() {
    try {
      const supplierResponse = await this.supplierService.getSupplierDetails(
        this.body,
      );
      const supplierModel = new SupplierAttributes(supplierResponse);
      const addressResponse = await this.addressService.getPartnerAddresses(1, this.body.uuid);
      const contactResponse = await this.contactService.getPartnerContacts(1, this.body.uuid);
      supplierModel.updateAddressAttributes(addressResponse);
      supplierModel.updateContacrAttributes(contactResponse);
      this.response.send(supplierModel);
    } catch (error) {
      const controllerError = new ErrorBuilder('controller', error);
      logger.info(controllerError.message);
      this.response.status(controllerError.code);
      this.response.end();
    }
  }

  async getSupplierSites() {
    try {
      const sitesResponse = await this.supplierService.getSupplierSites(
        this.body,
      );
      const siteRecords = sitesResponse.content.map(c => new SiteAttributes(c));
      this.response.send(siteRecords);
    } catch (error) {
      const controllerError = new ErrorBuilder('controller', error);
      logger.info(controllerError.message);
      this.response.status(controllerError.code);
      this.response.end();
    }
  }

  async getSupplierPayments() {
    try {
      const paymentsResponse = await this.supplierService.getSupplierPayments(
        this.body,
      );
      const paymentRecords = paymentsResponse.content.map(
        c => new PaymentAttributes(c),
      );
      this.response.send(paymentRecords);
    } catch (error) {
      const controllerError = new ErrorBuilder('controller', error);
      logger.info(controllerError.message);
      this.response.status(controllerError.code);
      this.respAonse.end();
    }
  }

  async getPartnerDetails() {
    try {
      const partnerResponse = await this.supplierService.getPartnerDetails(
        this.body,
      );
      const partnerModel = new SupplierAttributes(partnerResponse);
      this.response.send(partnerModel);
    } catch (error) {
      const controllerError = new ErrorBuilder('controller', error);
      logger.info(controllerError.message);
      this.response.status(controllerError.code);
      this.response.end();
    }
  }
}

module.exports = SupplierController;