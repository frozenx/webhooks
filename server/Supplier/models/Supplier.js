class SupplierModel {
  constructor(payload) {
    this.entityType = payload.entityType;
    this.uuid = payload.uuid;
    this.number = payload.number;
    this.name = payload.name;
    this.partnerTaxRegion = payload.partnerTaxRegion;
    this.taxOrganisationType = payload.taxOrganisationType;
    this.status = payload.status;
    this.siteInvoiceCurrency = payload.siteInvoiceCurrency;
    this.productCategory = payload.productCategory;
    this.subGroup = payload.subGroup;
  }
}

module.exports = SupplierModel;
