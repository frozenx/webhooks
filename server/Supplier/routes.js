const express = require('express');
const SupplierController = require('./controller');


const router = express.Router();


const handlers = {
  readSupplier: (req, res) => {
    const supplierController = new SupplierController(req, res);
    supplierController.readSupplier();
  },
  createSupplier: (req, res) => {
    const controller = new SupplierController(req, res);
    controller.createSupplier();
  },
  searchAndCreate: (req, res) => {
    const controller = new SupplierController(req, res);
    controller.readAndCreate();
  },
  getSupplierDetails: (req, res) => {
    const controller = new SupplierController(req, res);
    controller.getSupplierDetails();
  },
  getSupplierSites: (req, res) => {
    const controller = new SupplierController(req, res);
    controller.getSupplierSites();
  },
  getSupplierPayments: (req, res) => {
    const controller = new SupplierController(req, res);
    controller.getSupplierPayments();
  },
  getPartnerDetails: (req, res) => {
    const controller = new SupplierController(req, res);
    controller.getPartnerDetails(); 
  },
};

router.get('/supplier', handlers.readSupplier);
router.post('/supplier', handlers.createSupplier);
router.post('/searchAndCreate', handlers.searchAndCreate);
router.post('/supplier-details', handlers.getSupplierDetails);
router.post('/supplier-details/sites', handlers.getSupplierSites);
router.post('/supplier-details/payments', handlers.getSupplierPayments);
router.post('/partner-details-search', handlers.getPartnerDetails);

module.exports = {
  router,
  handlers,
};
