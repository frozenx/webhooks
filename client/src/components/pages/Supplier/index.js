import React, { Component } from "react";
import SupplierDetails from "../../atoms/SupplierDetails/";
import Loader from '../../atoms/Loader';
import EditSupplierDetails from "../../atoms/EditSupplierDetails/";
import {
  PrimaryWrapper,
  TabWrapper,
} from '../../../common/styles'

export class Supplier extends Component {
    state = {
      supplierDetails: true,
      editSupplierDetails: false,
      supplierData: null,
      supplierValues: {},
      editSupplierData: null,
      errorSupplierData:null
    }
  

   componentDidMount(){
     let id = this.props.id? this.props.id : 1234
     this.getData(id)
  }

  getData = async (id) => {
    try{
      const jsonRes = await fetch(`/tradingpartner/api/supplier?supplierNumber=${id}`);
      const data = await jsonRes.json();
      delete data.suppliers[0].id
      let supplierKey = Object.keys(data.suppliers[0]);
      this.setState({
        supplierData: supplierKey,
        supplierValues: data.suppliers[0],
        editSupplierData:data.suppliers[0]
      });
    } catch(err){
      // this.setState({
      //     errorSupplierData:err
      // });
    }
    
  }

  componentWillReceiveProps(nextProps, props) {
    if(nextProps.id != props.id) {
      this.getData(nextProps.id)
    }
  }
  _handleEditSupplierDetails = () => {
    this.setState({
      supplierDetails: false,
      editSupplierDetails: true,
      editSupplierData: this.state.supplierValues
    });
  };

  _handleSupplierChange = e => {
    let tempSupplierValues = {};
    const { name, value } = e.target;
    tempSupplierValues[name] = value;
    this.setState(prevState => ({
      supplierValues: { ...prevState.supplierValues, ...tempSupplierValues }
    }));
  };

  _handleSubmitSupplierDetails = (e) =>{
    e.preventDefault();
    this.setState({
      supplierDetails: true,
      editSupplierDetails: false,
      editSupplierData: this.state.supplierValues
    });
  }
  _handleCanselSupplierDetails = (e) =>{
    this.setState({
      supplierDetails: true,
      editSupplierDetails: false,
    });
  }

  render() {
    const {
      supplierDetails,
      editSupplierDetails,
      supplierData,
      editSupplierData
    } = this.state;

      const { activeTab } = this.state;
    if (!editSupplierData) {
      return (<div className="loader-parent-container"><Loader /></div>)
  } 
  else {
      return (
        <PrimaryWrapper>
           <TabWrapper>
            <div>
            {(supplierDetails && editSupplierData!=null) ? (
              <SupplierDetails
                handleEditSupplierDetails={
                  this._handleEditSupplierDetails
                }
                supplierData={supplierData}
                editSupplierData={editSupplierData}
              />
            ) : null}
            {editSupplierDetails ? (
              <EditSupplierDetails
                editSupplierData={editSupplierData}
                handleSupplierChange={e =>
                  this._handleSupplierChange(e)
                }
                handleSubmitSupplierDetails={
                  this._handleSubmitSupplierDetails
                }
                handleCanselSupplierDetails={this._handleCanselSupplierDetails}
                supplierData={supplierData}
              />
            ) : null}
            </div>
          </TabWrapper>
        </PrimaryWrapper>
      );
    
    }
    
    }
  }

export default Supplier;