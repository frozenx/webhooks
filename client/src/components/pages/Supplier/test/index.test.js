import React from "react";
import Supplier from "../";
import renderer from "react-test-renderer";
import SupplierDetails from "../../../atoms/SupplierDetails/";
import EditSupplierDetails from "../../../atoms/EditSupplierDetails/";

import Enzyme, { shallow, mount } from "enzyme";

describe("render Supplier component inside index file", () => {
  let Wrapper, w;
  let editSupplierData = {suppliers: [{
    supplierName: "Test Supplier Name",
    companiesHouseName: "Test Companies House Name",
    companyRegistrationNumber: "Test 123456",
    vatNumber: "Test 09876",
    id: 123,
  }]
}
  let supplierData = [
    "supplierName",
    "companiesHouseName",
    "companyRegistrationNumber",
    "vatNumber"
  ];

  let mockPreventDefault = jest.fn()
  let mockedEvent = {target: {}, preventDefault: mockPreventDefault }

  beforeEach(() => {
    Wrapper = shallow(<Supplier />);
    
  });

  it("renders the create supplier component", () => {
    expect(Wrapper).toMatchSnapshot();
  });

  it('should render the Supplier Details' , async () => {
   window.fetch = jest.fn().mockImplementation(() => ({
     status: 200,
     json: () => new Promise((resolve, reject) => {
       resolve(
         editSupplierData
       )
     })
   })) 
   const renderedComponent = await shallow(<Supplier />)
   await renderedComponent.update();
   expect(renderedComponent).toMatchSnapshot();
   expect(renderedComponent.find('SupplierDetails')).toBeTruthy();
  })

  it('should render the supplier details with default prop', () => {
    const wrapperWithNewProp = shallow(<Supplier id="4567"/>)
    expect(wrapperWithNewProp).toMatchSnapshot();
  })

  it('should call the getData when props got updates', () => {
    const wrapperWithDefaultProp = shallow(<Supplier />)
    const spyGetData = jest.spyOn(wrapperWithDefaultProp.instance(), 'getData');
    wrapperWithDefaultProp.setProps({id: 9080})
    expect(spyGetData).toHaveBeenCalled();
  })

  it('should trigger _handleEditSupplierDetails and Edit supplier details component should load', async (done) => {
    window.fetch = jest.fn().mockImplementation(() => ({
      status: 200,
      json: () => new Promise((resolve, reject) => {
        resolve(
          editSupplierData
        )
      })
    })) 
    const mountWrapper = await mount(<Supplier />)
    setTimeout(() => {
      mountWrapper.update();
      expect(mountWrapper.find('Supplier').length).toEqual(1);
      mountWrapper.find('.edit-supplier').simulate('click')
      expect(mountWrapper.find('EditSupplierDetails').length).toEqual(1)
      done()
    },0)
  })

  it('should trigger _handleSupplierChange ', async (done) => {
    window.fetch = jest.fn().mockImplementation(() => ({
      status: 200,
      json: () => new Promise((resolve, reject) => {
        resolve(
          editSupplierData
        )
      })
    })) 
    const mountWrapper = await mount(<Supplier />)
    setTimeout(() => {
      mountWrapper.update();
      mountWrapper.find('.edit-supplier').simulate('click')
      mountWrapper.find('form').simulate('change', {target: {name: 'some name', value: 'some value'}})
      // need to expect this thing not working with simulate      
      // expect(mountWrapper.instance()._handleSupplierChange).toHaveBeenCalled()
      done()
    },0)
  })

  it('should trigger _handleSubmitSupplierDetails ', async (done) => {
    let mockPreventDefault = jest.fn()
    let mockedEvent = {target: {}, preventDefault: mockPreventDefault }
    window.fetch = jest.fn().mockImplementation(() => ({
      status: 200,
      json: () => new Promise((resolve, reject) => {
        resolve(
          editSupplierData
        )
      })
    })) 
    const mountWrapper = await mount(<Supplier />)
    setTimeout(() => {
      mountWrapper.update();
      mountWrapper.find('.edit-supplier').simulate('click')
      mountWrapper.find('EditSupplierDetails').props().handleSubmitSupplierDetails({preventDefault: jest.fn()})
      // need to expect this thing not working with simulate      
      done()
    },0)
  })

  it('should trigger _handleCanselSupplierDetails ', async (done) => {
    let mockPreventDefault = jest.fn()
    let mockedEvent = {target: {}, preventDefault: mockPreventDefault }
    window.fetch = jest.fn().mockImplementation(() => ({
      status: 200,
      json: () => new Promise((resolve, reject) => {
        resolve(
          editSupplierData
        )
      })
    })) 
    const mountWrapper = await mount(<Supplier />)
    setTimeout(() => {
      mountWrapper.update();
      mountWrapper.find('.edit-supplier').simulate('click')
      mountWrapper.find('EditSupplierDetails').props().handleCanselSupplierDetails({preventDefault: jest.fn()})
      // need to expect this thing not working with simulate
      done()
    },0)
  })

});
