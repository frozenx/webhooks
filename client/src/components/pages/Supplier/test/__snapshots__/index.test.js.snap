// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Supplier component inside index file renders the create supplier component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Supplier />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader />,
      "className": "loader-parent-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader />,
        "className": "loader-parent-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`render Supplier component inside index file should render the Supplier Details 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Supplier />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <div>
          <SupplierDetails
            editSupplierData={
              Object {
                "companiesHouseName": "Test Companies House Name",
                "companyRegistrationNumber": "Test 123456",
                "supplierName": "Test Supplier Name",
                "vatNumber": "Test 09876",
              }
            }
            handleEditSupplierDetails={[Function]}
            supplierData={
              Array [
                "supplierName",
                "companiesHouseName",
                "companyRegistrationNumber",
                "vatNumber",
              ]
            }
          />
        </div>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          <SupplierDetails
            editSupplierData={
              Object {
                "companiesHouseName": "Test Companies House Name",
                "companyRegistrationNumber": "Test 123456",
                "supplierName": "Test Supplier Name",
                "vatNumber": "Test 09876",
              }
            }
            handleEditSupplierDetails={[Function]}
            supplierData={
              Array [
                "supplierName",
                "companiesHouseName",
                "companyRegistrationNumber",
                "vatNumber",
              ]
            }
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SupplierDetails
              editSupplierData={
                Object {
                  "companiesHouseName": "Test Companies House Name",
                  "companyRegistrationNumber": "Test 123456",
                  "supplierName": "Test Supplier Name",
                  "vatNumber": "Test 09876",
                }
              }
              handleEditSupplierDetails={[Function]}
              supplierData={
                Array [
                  "supplierName",
                  "companiesHouseName",
                  "companyRegistrationNumber",
                  "vatNumber",
                ]
              }
            />,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "editSupplierData": Object {
                "companiesHouseName": "Test Companies House Name",
                "companyRegistrationNumber": "Test 123456",
                "supplierName": "Test Supplier Name",
                "vatNumber": "Test 09876",
              },
              "handleEditSupplierDetails": [Function],
              "supplierData": Array [
                "supplierName",
                "companiesHouseName",
                "companyRegistrationNumber",
                "vatNumber",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htoDjs",
          "isStatic": true,
          "rules": Array [
            "
    margin: 20px 0px;
    display: inline-block;
    width: 100%;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htoDjs",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": true,
        "rules": Array [
          "
    margin: 0 auto;
    @media (max-width: 1259px){
        width: 100%;
    }
    @media (min-width: 1260px) {
        width: 1260px;
    }
    @media (min-width: 1512px){
        width: 1512px;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <div>
            <SupplierDetails
              editSupplierData={
                Object {
                  "companiesHouseName": "Test Companies House Name",
                  "companyRegistrationNumber": "Test 123456",
                  "supplierName": "Test Supplier Name",
                  "vatNumber": "Test 09876",
                }
              }
              handleEditSupplierDetails={[Function]}
              supplierData={
                Array [
                  "supplierName",
                  "companiesHouseName",
                  "companyRegistrationNumber",
                  "vatNumber",
                ]
              }
            />
          </div>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div>
            <SupplierDetails
              editSupplierData={
                Object {
                  "companiesHouseName": "Test Companies House Name",
                  "companyRegistrationNumber": "Test 123456",
                  "supplierName": "Test Supplier Name",
                  "vatNumber": "Test 09876",
                }
              }
              handleEditSupplierDetails={[Function]}
              supplierData={
                Array [
                  "supplierName",
                  "companiesHouseName",
                  "companyRegistrationNumber",
                  "vatNumber",
                ]
              }
            />
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SupplierDetails
                editSupplierData={
                  Object {
                    "companiesHouseName": "Test Companies House Name",
                    "companyRegistrationNumber": "Test 123456",
                    "supplierName": "Test Supplier Name",
                    "vatNumber": "Test 09876",
                  }
                }
                handleEditSupplierDetails={[Function]}
                supplierData={
                  Array [
                    "supplierName",
                    "companiesHouseName",
                    "companyRegistrationNumber",
                    "vatNumber",
                  ]
                }
              />,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "editSupplierData": Object {
                  "companiesHouseName": "Test Companies House Name",
                  "companyRegistrationNumber": "Test 123456",
                  "supplierName": "Test Supplier Name",
                  "vatNumber": "Test 09876",
                },
                "handleEditSupplierDetails": [Function],
                "supplierData": Array [
                  "supplierName",
                  "companiesHouseName",
                  "companyRegistrationNumber",
                  "vatNumber",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htoDjs",
            "isStatic": true,
            "rules": Array [
              "
    margin: 20px 0px;
    display: inline-block;
    width: 100%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htoDjs",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "rules": Array [
            "
    margin: 0 auto;
    @media (max-width: 1259px){
        width: 100%;
    }
    @media (min-width: 1260px) {
        width: 1260px;
    }
    @media (min-width: 1512px){
        width: 1512px;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`render Supplier component inside index file should render the supplier details with default prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Supplier
    id="4567"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader />,
      "className": "loader-parent-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader />,
        "className": "loader-parent-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
